src/puppetlabs/puppetdb/cli/generate.clj

```diff
@@ -0,0 +1,309 @@
+(ns puppetlabs.puppetdb.cli.generate
+  "Data Generation utility
+
+   This command-line tool can generate a base sampling of catalog, fact and
+   report files suitable for consumption by the PuppetDB benchmark utility.
+
+   Note that it is only necessary to generate a small set of initial sample
+   data since benchmark will permute per node differences. So even if you want
+   to benchmark 1000 nodes, you don't need to generate initial
+   catalog/fact/report json for 1000 nodes.
+
+   TODO:
+   * facts
+   * reports"
+  (:require
+    [clojure.set :as cset]
+    [clojure.string :as string]
+    [clojure.data.generators :as dgen]
+    [clojure.tools.namespace.dependency :as dep]
+    [puppetlabs.i18n.core :refer [trs]]
+    [puppetlabs.kitchensink.core :as kitchensink]
+    [puppetlabs.puppetdb.cheshire :as json]
+    [puppetlabs.puppetdb.cli.util :refer [exit run-cli-cmd]]
+    [puppetlabs.puppetdb.export :as export]
+    [puppetlabs.puppetdb.nio :refer [get-path]]
+    [puppetlabs.puppetdb.random :as rnd]
+    [puppetlabs.puppetdb.time :refer [now]]
+    [puppetlabs.puppetdb.utils :as utils])
+  (:import
+    [java.nio.file.attribute FileAttribute]
+    [java.nio.file Files]
+    [org.apache.commons.lang3 RandomStringUtils]))
+
+(defn pseudonym
+  "Generate a fictious but somewhat intelligible keyword name."
+  ([prefix ordinal]
+   (pseudonym prefix ordinal 6))
+  ([prefix ordinal length]
+   (let [mid-length (max (- length (count prefix) (count (str ordinal))) 1)]
+     (format "%s-%s-%s" prefix (rnd/random-pronouncable-word mid-length) ordinal))))
+
+(defn build-to-size
+  "Build up list of strings to approximately size bytes."
+  [size word-fn]
+  (let [builder (fn [v]
```

Comment:
This can be written using `loop` if you want to avoid the helper function.

```clj
(loop [v []]
  (if (< (reduce + (map count v)) size)
    (recur (conj v (word-fn)))
    v))
```

Reply:
That's nicer; fixed this and build-parameters below.